recipes/quickstart/agents/Agents_101/Tool_Calling_101.ipynb

```diff
@@ -0,0 +1,980 @@
+    "\n",
+    "This logic gets handled in the program and then the output is passed back to the model to answer the user"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Code interpreter\n",
+    "\n",
+    "With the correct prompt template, Llama model can output Python (as well as code in any-language that the model has been trained on)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 54,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Assistant: <|python_tag|>import math\n",
+      "\n",
+      "# Define the variables\n",
+      "monthly_investment = 400\n",
+      "interest_rate = 0.05\n",
+      "target_amount = 100000\n",
+      "\n",
+      "# Calculate the number of months it would take to reach the target amount\n",
+      "months = 0\n",
+      "current_amount = 0\n",
+      "while current_amount < target_amount:\n",
+      "    current_amount += monthly_investment\n",
+      "    current_amount *= 1 + interest_rate / 12  # Compound interest\n",
+      "    months += 1\n",
+      "\n",
+      "# Print the result\n",
+      "print(f\"It would take {months} months, approximately {months / 12:.2f} years, to reach the target amount of ${target_amount:.2f}.\")\n"
+     ]
+    }
+   ],
+   "source": [
+    "user_input = \"\"\"\n",
+    "\n",
+    "If I can invest 400$ every month at 5% interest rate, how long would it take me to make a 100k$ in investments?\n",
+    "\"\"\"\n",
+    "\n",
+    "print(\"Assistant:\", model_chat(user_input, sys_prompt=SYSTEM_PROMPT))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's validate the output by running the output from the model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 55,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "It would take 172 months, approximately 14.33 years, to reach the target amount of $100000.00.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Define the variables\n",
+    "monthly_investment = 400\n",
+    "interest_rate = 0.05\n",
+    "target_amount = 100000\n",
+    "\n",
+    "# Calculate the number of months it would take to reach the target amount\n",
+    "months = 0\n",
+    "current_amount = 0\n",
+    "while current_amount < target_amount:\n",
+    "    current_amount += monthly_investment\n",
+    "    current_amount *= 1 + interest_rate / 12  # Compound interest\n",
+    "    months += 1\n",
+    "\n",
+    "# Print the result\n",
+    "print(f\"It would take {months} months, approximately {months / 12:.2f} years, to reach the target amount of ${target_amount:.2f}.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3.2 Models Prompt Format"
```

Comment:
wonder if the title is closer to Custom Tool Calling? or something along those lines?